### ISSUE TITLE
Radio station spawning at intersection with wrong facing away from roads

### ISSUE THREAD

**Issue by RandDenner** [CleverRaven/Cataclysm-DDA#50938]
### Describe the bug

Even though there were two roads, mapgen seems to have become confused and not chosen either for facing.

### Steps To Reproduce

Absolutely default mapgen. But it's a station where the weather info for a quest generated.

### Expected behavior

It should choose a road to face.

### Screenshots

![image](https://user-images.githubusercontent.com/73523034/130302596-a28d141f-13e6-4be5-9e5e-1b69173425a5.png)

### Versions and configuration

- OS: Win10 64 bit
- Game Version: 0.F experimental cc50de2
- Graphics version: Tiles
- Ingame language: English
- Mods loaded: none

**Comment by RandDenner** [comment]
Ping: @actual-nh, will this count for confirmation?

Started a new world with 100% default settings and no mods.
Went to three different stations.
One at 4-way intersection, one at T intersection, one in T-intersection, but not on corner. All were rotated 90° counterclockwise from appropriate street facing front entrances.

4-way intersection:
![image](https://user-images.githubusercontent.com/73523034/130334666-3ed14759-b749-4b8b-bc9f-511d0c31e0d9.png)
(map)
![image](https://user-images.githubusercontent.com/73523034/130334670-ac786bdb-2a18-4ea6-874e-4113894d561b.png)

T-intersection:
![image](https://user-images.githubusercontent.com/73523034/130334674-36511178-2717-4aef-a7be-3a7711caa3f2.png)
(map)
![image](https://user-images.githubusercontent.com/73523034/130334685-3dd4e1bf-b829-41c5-a9d9-9111626565a0.png)

T-intersection, but not on a corner:
![image](https://user-images.githubusercontent.com/73523034/130334700-c4e34dfa-1bb7-46c2-98ad-a85c772ac3b6.png)
(map)
![image](https://user-images.githubusercontent.com/73523034/130334704-abb5ca47-27a9-469a-a625-7bcd9b184cc9.png)

**Comment by RandDenner** [comment]
Note: I believe I understand the problem with the orientations.
It seems it is dependent on the layout given in the file station_radio.json
I am manually rotating the map layout and transforming the coordinates section to the correct revisions.
I'll post a corrected version here when complete.

**Comment by RandDenner** [comment]
I just noticed churches are having the same issue with facing away from the street.
I'm planning to look at all buildings to ensure that they have correct street facings.

Can anyone explain to me why this is an issue?
Did there used to be a tag indicating the "front" building facing for a building template, and then code was changed and it was removed and so the code for building placement assumes now that the top of the layout (north) in the template is the street facing?
